FluentUI/src/main/java/com/microsoft/fluentui/contextualcommandbar/ContextualCommandBar.kt

```diff
@@ -0,0 +1,276 @@
+    private var dismissButtonContainer: LinearLayout? = null
+
+    // Dismiss button configurations
+    var showDismiss = false
+        set(value) {
+            field = value
+            setDismissButtonVisible(value)
+        }
+    @DrawableRes
+    var dismissIcon: Int = 0
+        set(value) {
+            if (value == 0) {
+                return
+            }
+
+            field = value
+            setDismissButtonVisible(showDismiss)
+            dismissButton?.setImageResource(dismissIcon)
+        }
+
+    var dismissListener: (() -> Unit)? = null
+    var itemClickListener: OnItemClickListener? = null
+
+    init {
+        removeAllViews()
+        addView(horizontalScrollView)
+        horizontalScrollView.addView(commandContainer)
+
+        with(commandContainer) {
+            orientation = LinearLayout.HORIZONTAL
+            gravity = Gravity.CENTER
+        }
+
+        with(horizontalScrollView) {
+            isVerticalScrollBarEnabled = false
+            isHorizontalScrollBarEnabled = false
+        }
+
+        attrs?.let {
+            val styledAttributes = context.theme.obtainStyledAttributes(
+                    it,
+                    R.styleable.ContextualCommandBar,
+                    0,
+                    0
+            )
+
+            try {
+                groupSpace = styledAttributes.getDimensionPixelSize(
+                        R.styleable.ContextualCommandBar_groupSpace,
+                        resources.getDimensionPixelSize(R.dimen.fluentui_contextual_command_bar_default_group_space)
+                )
+                itemSpace = styledAttributes.getDimensionPixelSize(
+                        R.styleable.ContextualCommandBar_itemSpace,
+                        resources.getDimensionPixelSize(R.dimen.fluentui_contextual_command_bar_default_item_space)
+                )
+                showDismiss = styledAttributes.getBoolean(
+                        R.styleable.ContextualCommandBar_showDismiss,
+                        false
+                )
+            } finally {
+                styledAttributes.recycle()
+            }
+
+            itemPaddingVertical = resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_default_item_padding_vertical
+            )
+            itemPaddingHorizontal = resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_default_item_padding_horizontal
+            )
+        }
+
+        setDismissButtonVisible(showDismiss)
+    }
+
+    private fun addDismissButton() {
+        // Container
+        dismissButtonContainer = LinearLayout(context).apply {
+            orientation = LinearLayout.HORIZONTAL
+        }
+        addView(dismissButtonContainer)
+        (dismissButtonContainer!!.layoutParams as LayoutParams).apply {
+            height = ViewGroup.LayoutParams.MATCH_PARENT
+            width = ViewGroup.LayoutParams.WRAP_CONTENT
+            gravity = Gravity.END
```

Comment:
likely no, but do we every envisage a scenario where a consumer might want the dismiss button on the leading side of the command groups (i.e. the left in LTR)?